kratos/solving_strategies/strategies/explicit_solving_strategy_runge_kutta.h

```diff
@@ -34,6 +36,224 @@ namespace Kratos
 ///@name Type Definitions
 ///@{
 
+/** @brief Butcher tableau for Runge-Kutta method.
+ *
+ * Contains all info necessary of a particular RK method.
+ * It specifies the coefficients of the particular Runge-Kutta method.
+ *
+ * @tparam: Derived:       A derived class that must contain the methods specified below.
+ * @tparam: TOrder:        The order of integration.
+ * @tparam: TSubstepCount: The number of sub-steps.
+ *
+ * Child class must provide methods:
+ * - static const MatrixType GenerateRKMatrix() -> Provides coefficients a_ij
+ * - static const VectorType GenerateWeights()  -> Provides coefficients b_i
+ * - static const VectorType GenerateNodes()    -> Provides coefficients c_i
+ * - static std::string Name()
+ */
+template<typename Derived, unsigned int TOrder, unsigned int TSubstepCount>
+class ButcherTableau
+{
+public:
+    static constexpr unsigned int Order() {return TOrder;}
+    static constexpr unsigned int SubstepCount() {return TSubstepCount; }
+
+    typedef array_1d<double, SubstepCount()> VectorType;
+    typedef std::array<array_1d<double, SubstepCount()-1>, SubstepCount()-1> MatrixType;
+
+    /**
+     * The runge kutta must perform for all substeps 1...N-1:
+     *
+     *  u^(i) = u^(i-1) + dt * A_ij*k_j
+     *
+     * This method computes the A_ij*k_j product
+     *
+     * @param SubstepIndex: The i in the formula (the row of the matrix)
+     * @param rK: The k in the formula (the reaction)
+     */
+    double MatrixReactionProduct(
```

Comment:
I think this operation must be done at the strategy level. For me the `ButcherTableau` must only contain "information". Also note that as it is right now you're doing the BCs intermediate update in the strategy while the non-fixed intermediate update values are done in here.

Reply:
Agreed. Changing it.